src/ucs/arch/x86_64/atomic.h

```diff
@@ -50,4 +50,17 @@
         return prev; \
     }
 
+#define UCS_DEFINE_ATOMIC_BOOL_CSWAP(wordsize, suffix) \
```

Comment:
Compilers have been doing really good job (with the exception of non standard 128b) on AMO. Do you really need asm implementation here ?

Reply:
it's a good question - i simply copied the implementation that was already there for ucs_atomic_cswap. 
If you believe there is a need to switch the current implementation with a compiler's AMO, please open a separate pull request.
from a brief look at https://gcc.gnu.org/wiki/Atomic , looks like the atomics there should be supported on x86_64

Reply:
@yosefe maybe you will remember the reason why for x86_64 we have chosen hand coded AMOs ?

Reply:
maybe it was not available on all compilers.. in general i'm ok with removing these and always using the generic version, but this is a bigger change for another PR

Reply:
@shamisp @yosefe would you like me to open a seperate PR for this? or should i leave it for someone else?

Reply:
> would you like me to open a seperate PR for this?

IMO no need for now

Reply:
I think I may remember why.... build-in amo operations have memory barrier. This implementation does not have  barrier with the assumption that upper layer (aka SHMEM programming model, etc.) will require explicit synchronization.  I'm okay to merge this as it is. It will be useful to objdump gcc generated code and compare it with handcrafted instructions.